packages/mui-material/src/Button/Button.js

```diff
@@ -600,7 +600,7 @@ const Button = React.forwardRef(function Button(inProps, ref) {
       focusVisibleClassName={clsx(classes.focusVisible, focusVisibleClassName)}
       ref={ref}
       type={type}
-      id={id}
+      id={loadingIndicator ? id : undefined}
```

Comment:
I'm changing this logic, otherwise a user passed `id` prop won't be applied. I'll also add a test.